src/memory_server.ml

```diff
@@ -278,22 +298,45 @@ let calculate_boot_time_host_free_memory () =
       pages_to_kib (Int64.of_nativeint boot_time_host_free_pages) in
     Int64.mul 1024L boot_time_host_free_kib
 
+let calc_constant_boot_time_host_free_memory constant_count interval =
+  let tolerance = 10L in
+  debug "Check boot time host free memory: constant-count=%d check-interval=%f(seconds)" constant_count interval;
+  let rec calc_constant last same_count =
+    let free_memory = calculate_boot_time_host_free_memory () in
+    debug "Calculating boot time host free memory: %Ld" free_memory;
+    let same_count = match last with
+        | Some old_free_memory when Int64.abs(Int64.sub free_memory old_free_memory) <= tolerance -> same_count + 1
+        | _ -> 1
+    in
+    if same_count >= constant_count then begin
+      debug "Boot time host free memory constant value: %Ld" free_memory;
+      free_memory
+    end else begin
+      Thread.delay interval;
+      calc_constant (Some free_memory) same_count
+    end
+  in
+  calc_constant None 0
+
 (* Read the free memory on the host and record this in the db. This is used *)
 (* as the baseline for memory calculations in the message forwarding layer. *)
 let record_boot_time_host_free_memory () =
   if not (Unixext.file_exists initial_host_free_memory_file) then begin
     try
-      let free_memory = calculate_boot_time_host_free_memory () in
+      let free_memory = calc_constant_boot_time_host_free_memory !Squeeze.boot_time_host_free_memory_constant_count !Squeeze.boot_time_host_free_memory_check_interval in
```

Comment:
I think the name should be updated here as well

Reply:
Yes